buildSrc/src/test/java/org/elasticsearch/gradle/DistributionDownloadPluginIT.java

```diff
@@ -0,0 +1,161 @@
+
+    public void testCurrent() throws Exception {
+        String projectName = ":distribution:archives:linux-tar";
+        assertFileDistro(VersionProperties.getElasticsearch(), "archive", "linux", null, null,
+            "tests.local_distro.config", "default",
+            "tests.local_distro.project", projectName);
+        assertExtractedDistro(VersionProperties.getElasticsearch(), "archive", "linux", null, null,
+            "tests.local_distro.config", "default",
+            "tests.local_distro.project", projectName);
+    }
+
+    public void testBwc() throws Exception {
+        assertFileDistro("1.1.0", "archive", "linux", null, null,
+            "tests.local_distro.config", "linux-tar",
+            "tests.local_distro.project", ":distribution:bwc:minor",
+            "tests.current_version", "2.0.0");
+        assertExtractedDistro("1.1.0", "archive", "linux", null, null,
+            "tests.local_distro.config", "linux-tar",
+            "tests.local_distro.project", ":distribution:bwc:minor",
+            "tests.current_version", "2.0.0");
+    }
+
+    public void testReleased() throws Exception {
+        WireMockServer wireMock = new WireMockServer(0);
+        try {
+            final byte[] filebytes;
+            try (InputStream stream =
+                     Files.newInputStream(Paths.get("src/testKit/distribution-download/distribution/files/fake_elasticsearch.zip"))) {
+                filebytes = stream.readAllBytes();
+            }
+            String urlPath = "/downloads/elasticsearch/elasticsearch-1.0.0-windows-x86_64.zip";
+            wireMock.stubFor(head(urlEqualTo(urlPath)).willReturn(aResponse().withStatus(200)));
+            wireMock.stubFor(get(urlEqualTo(urlPath)).willReturn(aResponse().withStatus(200).withBody(filebytes)));
+            wireMock.start();
+
+            assertFileDistro("1.0.0", "archive", "windows", null, null,
+                "tests.download_service", wireMock.baseUrl());
+            assertExtractedDistro("1.0.0", "archive", "windows", null, null,
+                "tests.download_service", wireMock.baseUrl());
+        } catch (Exception e) {
+            // for debugging
+            System.err.println("missed requests: " + wireMock.findUnmatchedRequests().getRequests());
+            throw e;
+        } finally {
+            wireMock.stop();
+        }
+    }
+
+    private void assertFileDistro(String version, String type, String platform, String flavor, Boolean bundledJdk,
+                                  String... sysProps) throws IOException {
+        List<String> finalSysProps = new ArrayList<>();
+        addDistroSysProps(finalSysProps, version, type, platform, flavor, bundledJdk);
+        finalSysProps.addAll(Arrays.asList(sysProps));
+        runBuild(":subproj:getDistro", result -> {
+                Matcher matcher = DISTRO_FILE_LOGLINE.matcher(result.getOutput());
```

Comment:
FWIW, a very common pattern in Gradle integration tests is to do these assertions in a task action in the build. That way your test just tests that a particular task ran successfully and you can avoid all this awkward business with passing info between the build and the test with raw output that needs to be parsed with regular expressions.

All this could be replaced with `assert file("${elasticsearch_distributions.test_distro}").exists()` in the task action.